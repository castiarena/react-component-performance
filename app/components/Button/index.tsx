import styled, { css } from 'styled-components';

interface ButtonProps {
    kind?: 'primary' | 'secondary';
}

const Button = styled.button<ButtonProps>`
    border: none;
    padding: ${({ theme }) => theme.spacing.small} ${({ theme }) => theme.spacing.large};
    border-radius: ${({ theme }) => theme.borderRadius.large};
    ${({ theme, kind = 'primary' }) => {
        let color = theme.palette.primary;
        if (kind === 'secondary') {
            color = theme.palette[kind];
        }
        return css`
          background-color: ${color.main};
          color: ${color.contrast};
          
          &:hover {
            cursor: pointer;
            background-color: ${color.hover};
          }
          &:active, &:focus {
            outline: none;
          }
        `;
    }};
`;

export default Button;
