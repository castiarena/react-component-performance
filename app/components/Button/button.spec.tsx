import React from 'react';
import 'jest-styled-components';
import Button from '.';
import renderWithTheme from '../../testing-tools/renderWithTheme';

describe('Button test suite', () => {
    it('should render a primary button as default', () => {
        const { asFragment } = renderWithTheme(
            <Button>default button</Button>,
        );
        const fragment = asFragment();

        expect(fragment).toMatchSnapshot();
    });

    it('should render a primary button as default', () => {
        const { asFragment } = renderWithTheme(
            <Button kind="secondary">secondary Button</Button>,
        );
        const fragment = asFragment();

        expect(fragment).toMatchSnapshot();
    });
});
