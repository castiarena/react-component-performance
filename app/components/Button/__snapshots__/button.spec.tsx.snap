// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test suite should render a primary button as default 1`] = `
<DocumentFragment>
  .c0 {
  border: none;
  padding: 16px 30px;
  border-radius: 24px;
  background-color: #726a95;
  color: #fff;
}

.c0:hover {
  cursor: pointer;
  background-color: #8d83b8;
}

.c0:active,
.c0:focus {
  outline: none;
}

<button
    class="c0"
  >
    default button
  </button>
</DocumentFragment>
`;

exports[`Button test suite should render a primary button as default 2`] = `
<DocumentFragment>
  .c0 {
  border: none;
  padding: 16px 30px;
  border-radius: 24px;
  background-color: #709fb0;
  color: #fff;
}

.c0:hover {
  cursor: pointer;
  background-color: #84bcd0;
}

.c0:active,
.c0:focus {
  outline: none;
}

<button
    class="c0"
    kind="secondary"
  >
    secondary Button
  </button>
</DocumentFragment>
`;
